src/sig/slh_dsa/patches/hash_shims.patch

```diff
@@ -0,0 +1,110 @@
+-  sha2_256_final(&sha, h);
+-}
+-
+ void sha2_224(uint8_t *h, const void *m, size_t m_sz)
+ {
+   sha2_256_t sha;
+diff --git a/src/sig/slh_dsa/slh_dsa_c/sha2_512.c b/src/sig/slh_dsa/slh_dsa_c/sha2_512.c
+index 12a64649..1b7010ae 100644
+--- a/src/sig/slh_dsa/slh_dsa_c/sha2_512.c
++++ b/src/sig/slh_dsa/slh_dsa_c/sha2_512.c
+@@ -317,25 +317,6 @@ void sha2_512_final_len(sha2_512_t *sha, uint8_t *h, size_t h_sz)
+ }
+ 
+ /* public single-call interfaces */
+-
+-void sha2_512(uint8_t *h, const void *m, size_t m_sz)
+-{
+-  sha2_512_t sha;
+-
+-  sha2_512_init(&sha);
+-  sha2_512_update(&sha, m, m_sz);
+-  sha2_512_final(&sha, h);
+-}
+-
+-void sha2_384(uint8_t *h, const void *m, size_t m_sz)
+-{
+-  sha2_384_t sha;
+-
+-  sha2_384_init(&sha);
+-  sha2_512_update(&sha, m, m_sz);
+-  sha2_512_final_len(&sha, h, 48);
+-}
+-
+ void sha2_512_224(uint8_t *h, const void *m, size_t m_sz)
+ {
+   sha2_512_t sha;
+diff --git a/src/sig/slh_dsa/slh_dsa_c/sha2_api.h b/src/sig/slh_dsa/slh_dsa_c/sha2_api.h
+index 1128e591..367b5e4f 100644
+--- a/src/sig/slh_dsa/slh_dsa_c/sha2_api.h
++++ b/src/sig/slh_dsa/slh_dsa_c/sha2_api.h
+@@ -16,21 +16,13 @@ extern "C"
+ #include <stddef.h>
+ #include <stdint.h>
+ #include "cbmc.h"
++#include "sha2.h"
+ 
+   /* === Single-call hash wrappers === */
+ 
+   /* SHA2-224: Compute 28-byte hash to "md" from "m" which has "m_sz" bytes. */
+   void sha2_224(uint8_t *md, const void *m, size_t m_sz);
+ 
+-  /* SHA2-256: Compute 32-byte hash to "md" from "m" which has "m_sz" bytes. */
+-  void sha2_256(uint8_t *md, const void *m, size_t m_sz);
+-
+-  /* SHA2-384: Compute 48-byte hash to "md" from "m" which has "m_sz" bytes. */
+-  void sha2_384(uint8_t *md, const void *m, size_t m_sz);
+-
+-  /* SHA2-512: Compute 64-byte hash to "md" from "m" which has "m_sz" bytes. */
+-  void sha2_512(uint8_t *md, const void *m, size_t m_sz);
+-
+   /* SHA2-512/224: Compute 64-byte hash to "md" from "m" which has "m_sz" */
+   /* bytes. */
+   void sha2_512_224(uint8_t *md, const void *m, size_t m_sz);
+diff --git a/src/sig/slh_dsa/slh_dsa_c/sha2_256.c b/src/sig/slh_dsa/slh_dsa_c/sha2_256.c
+index ef83b8d4..5ed137d8 100644
+--- a/src/sig/slh_dsa/slh_dsa_c/sha2_256.c
++++ b/src/sig/slh_dsa/slh_dsa_c/sha2_256.c
+@@ -249,14 +249,3 @@ void sha2_256_final_len(sha2_256_t *sha, uint8_t *h, size_t h_sz)
+ 
+   memcpy(h, sha->s, h_sz);
+ }
+-
+-/* SHA-224/256 public single-call interfaces */
+-
+-void sha2_224(uint8_t *h, const void *m, size_t m_sz)
+-{
+-  sha2_256_t sha;
+-
+-  sha2_224_init(&sha);
+-  sha2_224_update(&sha, m, m_sz);
+-  sha2_224_final(&sha, h);
+-}
+diff --git a/src/sig/slh_dsa/slh_dsa_c/sha2_api.h b/src/sig/slh_dsa/slh_dsa_c/sha2_api.h
+index 367b5e4f..1e948cf8 100644
+--- a/src/sig/slh_dsa/slh_dsa_c/sha2_api.h
++++ b/src/sig/slh_dsa/slh_dsa_c/sha2_api.h
+@@ -20,9 +20,6 @@ extern "C"
+ 
+   /* === Single-call hash wrappers === */
+ 
+-  /* SHA2-224: Compute 28-byte hash to "md" from "m" which has "m_sz" bytes. */
```

Comment:
If this PR aims to remove all local SHA2 code and replace it with calls to `liboqs` common code, why does it look like sha2-512 is not being patched in the same way as is sha2-224?